app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,13 @@
+class WelcomeController < ApplicationController
+
+  def index
+    swanson_quote_service = SwansonQuoteService.new
+    puts swanson_quote_service.to_yaml
+    if swanson_quote_service.quote.nil?
+      @quote = {quote: '', rating: ''}
+    else
+      swanson_quote = SwansonQuote.find_or_create_by(quote: swanson_quote_service.quote)
+      @quote = {quote: swanson_quote.quote, rating: swanson_quote.quote_ratings.get_avg, did_rate: swanson_quote.quote_ratings.did_rate?(request.remote_ip) }
+    end
+  end
```

Comment:
All of this could be done in your Services class to make the controller more 'dumb'.

Then you could have done something like this.

response = SwansonQuoteService.new
if response.success?
# render json
else
#render error